shadows/framework/src/main/java/org/robolectric/shadows/ShadowCountDownTimer.java

```diff
@@ -1,60 +1,100 @@
 package org.robolectric.shadows;
 
+import static org.robolectric.util.reflector.Reflector.reflector;
+
 import android.os.CountDownTimer;
 import org.robolectric.annotation.Implementation;
 import org.robolectric.annotation.Implements;
 import org.robolectric.annotation.RealObject;
-import org.robolectric.shadow.api.Shadow;
-import org.robolectric.util.ReflectionHelpers.ClassParameter;
+import org.robolectric.util.reflector.Accessor;
+import org.robolectric.util.reflector.Direct;
+import org.robolectric.util.reflector.ForType;
 
 @Implements(CountDownTimer.class)
 public class ShadowCountDownTimer {
-  private boolean started;
-  private long countDownInterval;
-  private long millisInFuture;
-
-  @RealObject CountDownTimer countDownTimer;
+  static final String PROPERTY_USE_REAL_IMPL = "robolectric.useRealCountDownTimer";
 
-  @Implementation
-  protected void __constructor__(long millisInFuture, long countDownInterval) {
-    this.countDownInterval = countDownInterval;
-    this.millisInFuture = millisInFuture;
-    this.started = false;
-    Shadow.invokeConstructor(
-        CountDownTimer.class,
-        countDownTimer,
-        ClassParameter.from(long.class, millisInFuture),
-        ClassParameter.from(long.class, countDownInterval));
-  }
+  private boolean started;
+  private boolean cancelled;
```

Comment:
There is already a field `mCancelled` in CountDownTimer. We should just use that instead of duplicating it here: 

https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/core/java/android/os/CountDownTimer.java;l=76